runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1871,16 +1946,17 @@ private void schedulePeriodicGlobalConfigRequests() {
     LOG.info("windmillServerStub is now ready");
 
     // Now start a thread that periodically refreshes the windmill service endpoint.
-    globalConfigRefreshTimer = new Timer("GlobalConfigRefreshTimer");
-    globalConfigRefreshTimer.schedule(
-        new TimerTask() {
+    globalConfigRefreshTimer = executorSupplier.get();
+    globalConfigRefreshTimer.scheduleWithFixedDelay(
+        new Runnable() {
```

Comment:
Again this could be this::getGlobalConfig

Reply:
Done